crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -734,37 +756,68 @@ impl RealRibosome {
         &self,
         call_context: CallContext,
         invocation: &I,
-        zome: &Zome,
         to_call: &FunctionName,
         module: Arc<Module>,
     ) -> Result<Option<ExternIO>, RibosomeError> {
         if module.info().exports.contains_key(to_call.as_ref()) {
             // there is a callback to_call and it is implemented in the wasm
             // it is important to fully instantiate this (e.g. don't try to use the module above)
             // because it builds guards against memory leaks and handles imports correctly
-            let (instance_with_store, context_key) = self.instance_with_store(call_context)?;
+            let (instance_with_store, context_key) =
+                self.instance_with_store(call_context.clone())?;
 
             let result: Result<ExternIO, RuntimeError>;
             {
                 let instance = instance_with_store.instance.clone();
                 let mut store_lock = instance_with_store.store.lock();
                 let mut store_mut = store_lock.as_store_mut();
+                set_remaining_points(&mut store_mut, instance.as_ref(), WASM_METERING_LIMIT);
                 result = holochain_wasmer_host::guest::call(
                     &mut store_mut,
-                    instance,
+                    instance.clone(),
                     to_call.as_ref(),
                     // be aware of this clone!
                     // the whole invocation is cloned!
                     // @todo - is this a problem for large payloads like entries?
                     invocation.to_owned().host_input()?,
                 );
+                let points_used = match get_remaining_points(&mut store_mut, instance.as_ref()) {
+                    MeteringPoints::Remaining(points) => WASM_METERING_LIMIT - points,
+                    MeteringPoints::Exhausted => WASM_METERING_LIMIT,
+                };
+                self.usage_meter.add(
+                    points_used,
+                    &[
+                        opentelemetry_api::KeyValue::new(
+                            "dna",
+                            self.dna_file.dna().hash.to_string(),
+                        ),
+                        opentelemetry_api::KeyValue::new(
+                            "zome",
+                            call_context.zome().zome_name().to_string(),
+                        ),
+                        opentelemetry_api::KeyValue::new("fn", to_call.to_string()),
+                        opentelemetry_api::KeyValue::new(
+                            "chain",
```

Comment:
Is chain the clearest name for the agent public key? I'd probably prefer just `agent`?

These labels should also be included in the docs please